src/ICSharpCode.SharpZipLib/Core/PathUtils.cs

```diff
@@ -15,48 +15,9 @@ public static class PathUtils
 		/// <remarks>Unlike the <see cref="System.IO.Path"/> class the path isn't otherwise checked for validity.</remarks>
```

Comment:
Referring to the comment about not checking for validity - I can't recall if this is ever called on reading a file or just creating? - are there any situations where GetPathRoot might throw on .NET 4.x if it thinks something is invalid (e.g. same sort of situation as #341)

Reply:
Yeah, you are probably right. I'll add a test for this and fix it if it triggers anything.
If it does, it is probably better to use another code path for cleaning "input" file names, rather than sharing it with the "output" ones.

Reply:
The results are in!
![image](https://user-images.githubusercontent.com/807383/111062261-12388c00-84a8-11eb-839b-7e0baf8565e7.png)
The usual reserved characters for file names in windows are `\ / : * ? " < > |`, but out of those, only `< > | "` are listed in `Path.GetInvalidPathChars()` and throws on net46.
I guess the easiest way to solve it is to just replace those chars with `_`?

Reply:
Added an optional `replaceInvalidChars` to the arguments as well, since the work is already done, but per default it returns the stripped path, **including** any invalid path characters.

Reply:
I've forgotten all the details offhand about how DropPathRoot interacts with the other name transform/cleaning stuff, but looking at the comment about long paths reminded me of the ```\\?\``` long path syntax on Windows, the handling of which might be another reason to leave it up to the built in functions instead of doing things manually

Reply:
It's used in `WindowsNameTransform` (for zip -> windows file name), `PathNameTransform` and `ZipFileTransform` for `ZipOutputStream` and `FastZip`/`ZipEntryFactory` respectively.